### PR TITLE
Split worker-side monitoring message code based on purpose

### PR DESCRIPTION
**Pull request by benclifford**
Prior to this patch, monitor() performed two functions, selected by a boolean
parameter:

i) It sends a task running message, which is used to populate the task_try_running
database field.

ii) It performs periodic monitoring of the process tree that is executing the
task.

These two code paths are fairly clearly separate, and do not need to both be in
monitor.

This PR is preparation for future PRs which allow worker-side monitoring to
report task start time without forking a resource monitoring process.

## Type of change

- Code maintentance/cleanup

### CODE REVIEW

File: parsl/monitoring/monitoring.py [hunk 1]
```diff
@@ -527,40 +523,47 @@ def router_starter(comm_q: "queue.Queue[Union[Tuple[int, int], str]]",
...
-        radio.send(msg)
-        return
-
-    import psutil
-    import logging
-
     format_string = "%(asctime)s.%(msecs)03d %(name)s:%(lineno)d [%(levelname)s]  %(message)s"
     logging.basicConfig(filename='{logbase}/monitor.{task_id}.{pid}.log'.format(
-        logbase="/tmp", task_id=task_id, pid=pid), level=logging_level, format=format_string)
+        logbase="/tmp", task_id=task_id, pid=pid), level=logging.DEBUG, format=format_string)
```
Comment (ZhuozhaoLi) [inline parsl/monitoring/monitoring.py:566]: Is it intentionally changed to debug logging here? if yes, maybe should also remove logging_level, as it is not used anywhere now.

Comment (benclifford) [reply parsl/monitoring/monitoring.py:566]: no. i'll change it back

**Comment by ZhuozhaoLi** [comment]
Is this a new CI test error?

**Comment by benclifford** [comment]
this error
```
Exception ignored in: <function _after_at_fork_weak_calls at 0x7f553cadbea0>
Traceback (most recent call last):
  File "/opt/python/3.7.1/lib/python3.7/logging/__init__.py", line 269, in _after_at_fork_weak_calls
    _at_fork_weak_calls('release')
  File "/opt/python/3.7.1/lib/python3.7/logging/__init__.py", line 254, in _at_fork_weak_calls
    for instance in _at_fork_acquire_release_weakset:
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/_weakrefset.py", line 60, in __iter__
    for itemref in self.data:
RuntimeError: Set changed size during iteration
sss
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
The build has been terminated
```
happens often in our CI.

I haven't spend much time debugging it, so I'm unclear what is going on.

**Review by ZhuozhaoLi — APPROVED** [review]

**Pull request merged**